tests/test_indeed.py

```diff
@@ -0,0 +1,245 @@
+            except:
+                assert False
+        assert True
+
+
+# test the process of fetching the link to a job
+
+    def test_get_link(self, setup_scraper, search_terms_config):
+        scraper = setup_scraper('indeed')
+        job_soup_list = scraper['job_list']
+        job = scraper['job_keys']
+        provider = scraper['job_provider']
+        provider.search_terms = search_terms_config
+        for soup in job_soup_list:
+            try:
+                job['id'] = provider.get_id(soup)
+                job['link'] = provider.get_link(job['id'])
+            except AttributeError:
+                continue
+            if(0 < len(job['link'])):
+                assert True
+                return
+
+        assert False
+
+
+# test the process of fetching the blurb from a job
+
+    def test_get_blurb_with_delay(self, setup_scraper, search_terms_config):
+        """
+        Tests whether the process of fetching blurb data is working.
+        """
+        scraper = setup_scraper('indeed')
+        provider = scraper['job_provider']
+        job_soup_list = scraper['job_list']
+        job = scraper['job_keys']
+        provider.search_terms = search_terms_config
+        for soup in job_soup_list:
+            try:
+                job['id'] = provider.get_id(soup)
+                job['link'] = provider.get_link(job['id'])
+                res_job, html = provider.get_blurb_with_delay(job, delay_alg(
+                    len(job_soup_list), provider.delay_config)[0])
+                provider.parse_blurb(job, html)
+            except AttributeError:
+                continue
+            if(0 < len(job['blurb'])):
+                assert True
+                return
+
+        assert False
+
+
+         
+    def test_search_joblink_for_blurb(self, setup_scraper, search_terms_config):
+        """
+        Tests whether the process of fetching blurb data is working.
+        This test assumes that no delay configuration has been set.
+        """
+        scraper = setup_scraper('indeed')
+        provider = scraper['job_provider']
+        job_soup_list = scraper['job_list']
+        job = scraper['job_keys']
+        provider.delay_config = None
+        provider.search_terms = search_terms_config
+        for soup in job_soup_list:
+            try:
+                job['id'] = provider.get_id(soup)
+                job['link'] = provider.get_link(job['id'])
+                provider.search_joblink_for_blurb(job)
+            except AttributeError:
+                continue
+            if(0 < len(job['blurb'])):
+                assert True
+                return
+
+        assert False
+
+
+    # Test the entire integration
+
+    def test_scrape(self, init_scraper, monkeypatch, 
+    search_terms_config, get_number_of_pages_mock):
+        # ensure that we don't scrape more than one page
+        monkeypatch.setattr(
+            Indeed, 'get_number_of_pages', get_number_of_pages_mock)
```

Comment:
better to avoid monkeypatch and use `mocker`
```
def test_scrape(self, mocker, init_scraper, search_terms_config, 
                get_number_of_pages_mock):
        # ensure that we don't scrape more than one page
        mocker.patch('jobfunnel.indeed.Indeed.get_number_of_pages', return_value=1)
...
```

`mocker` is the best part of pytest I think, you can always access it as a fixture :)